pkg/network/proxy/proxy.go

```diff
@@ -312,49 +314,58 @@ func (proxy *OsdnProxy) OnEndpointsAdd(ep *corev1.Endpoints) {
 	proxy.Lock()
 	defer proxy.Unlock()
 
-	pep := &proxyEndpoints{ep, proxy.endpointsBlocked(ep)}
-	proxy.allEndpoints[ep.UID] = pep
-	if !pep.blocked {
-		proxy.baseProxy.OnEndpointsAdd(ep)
+	labels := ep.GetLabels()
+	if _, ok := labels[IsHeadLessService]; ok {
+		pep := &proxyEndpoints{ep, proxy.endpointsBlocked(ep)}
```

Comment:
Don't indent the whole function body; return early in the no-op case instead.

And the check is backwards anyway isn't it? (We want to process the endpoints if it *doesn't* have the annotation, and skip processing if it does.)